tests/main/interfaces-snap-interfaces-requests-control/task.yaml

```diff
@@ -0,0 +1,90 @@
+    # fact all the test does is pipe the output to jq
+    PYTHONIOENCODING: utf-8
+
+prepare: |
+    snap install --edge jq
+
+execute: |
+    "$TESTSTOOLS"/snaps-state install-local api-client
+    echo "The snap-interfaces-requests-control plug on the api-client snap is initially disconnected"
+    snap connections api-client | MATCH "snap-interfaces-requests-control +api-client:snap-interfaces-requests-control +- +-"
+    echo "Connect the snap-interfaces-requests-control plug"
+    snap connect api-client:snap-interfaces-requests-control
+
+    echo "Check snap can access interfaces-requests-prompt and interfaces-requests-rule-update notices under /v2/notices"
+    api-client --socket /run/snapd-snap.socket "/v2/notices?types=interfaces-requests-prompt" | jq '."status-code"' | MATCH '^200$'
+    api-client --socket /run/snapd-snap.socket "/v2/notices?types=interfaces-requests-rule-update" | jq '."status-code"' | MATCH '^200$'
+    api-client --socket /run/snapd-snap.socket "/v2/notices" | jq '."status-code"' | MATCH '^200$'
+    echo "But not other notice types"
+    api-client --socket /run/snapd-snap.socket "/v2/notices?types=change-update,warning" | jq '."status-code"' | MATCH '^403$'
+
+    echo "Check snap can access system info via /v2/system-info"
+    api-client --socket /run/snapd-snap.socket "/v2/system-info" | jq '."status-code"' | MATCH '^200$'
+
+    echo "Ensure AppArmor Prompting experimental feature is marked as enabled"
+    snap set system experimental.apparmor-prompting=true
+
+    echo 'Check "apparmor-prompting" is shown as enabled in /v2/system-info'
+    api-client --socket /run/snapd-snap.socket "/v2/system-info" | jq '."result"."features"."apparmor-prompting"."enabled"' | MATCH '^true$'
+
+    # TODO: remove shellcheck exception
+    #shellcheck disable=SC2034
+    EXPECTED_HTTP_CODE="200"
+    if api-client --socket /run/snapd-snap.socket "/v2/system-info" | jq '."result"."features"."apparmor-prompting"."supported"' | MATCH '^false$' ; then
+        # AppArmor prompting isn't supported, so rules and prompts backends are
+        # not active, and will return InternalError (500). We can at least check
+        # that we receive an InternalError instead of Forbidden (403).
+        echo "Prompting is not supported, so check that we get InternalError (500) instead of Forbidden (403)"
+        # TODO: remove shellcheck exception
+        #shellcheck disable=SC2034
+        EXPECTED_HTTP_CODE="500"
+    fi
+
+    # TODO: update once the prompting API is landed in snapd master
+    # echo "Check snap can access prompts via /v2/interfaces/requests/prompts"
+    # api-client --socket /run/snapd-snap.socket "/v2/interfaces/requests/prompts" | jq '."status-code"' | MATCH '^'"$EXPECTED_HTTP_CODE"'$'
+    # echo "Check snap can access a single prompt via /v2/interfaces/requests/prompts/<ID>"
+    # TODO: include the "home" interface and create a request prompt by attempting to list contents of $HOME
```

Comment:
I presume all this is coming once the internals are ready.

Reply:
Yes, I'll wire this up once I rebase the chain in a bit. Fun times.